spec/integration/_dev/deploy/agent/custom-agent.spec.yml

```diff
@@ -14,7 +14,7 @@ spec:
     services:
       description: Service list
       type: object
-      additionalProperties: false
+      additionalProperties: true
```

Comment:
It'd be nice if we could be more restrictive here. At least to apply the same restrictions as for the `docker-custom-agent` service.

In theory an schema can be defined here, could you try if this works?
```suggestion
      additionalProperties:
        description: Additional services
        type: object
        not:
          anyOf:
            - required:
                -hostname
```

If this works, please try to add a common definition in this file so it is reused along services. It would be something like this:
```
##
## Describes the specification for variations available when deploying a package's
## integration service(s).
##
spec:
  # Everything under here follows JSON schema (https://json-schema.org/), written as YAML for readability
  type: object
  additionalProperties: false
  definitions:
    service:
      type: object
        not:
          anyOf:
            - required:
               - hostname
  properties:
    version:
      description: Docker Compose version
      type: string
      pattern: '^2.3$'
    services:
      description: Service list
      type: object
      additionalProperties:
        description: Additional services
        $ref: #/definitions/service
      properties:
        docker-custom-agent:
          description: Custom agent service definition
          $ref: #/definitions/service
      required:
        - docker-custom-agent
  required:
    - version
    - services
```

Reply:
@jsoriano , Thanks a lot for the suggestion. 
i had to make some minor adjustments to make this work. 
Please review.

Reply:
Yeah, looks good :+1:  Could you please also add some sample of this configuration in the test packages?

Reply:
Added the test cases. Test case output added

```
agikthomas@service-integration-dev-idc-01:~/go/src/github.com/elastic/package-spec$ make test | grep 'deploy_custom_agent'
=== RUN   TestValidateFile/deploy_custom_agent_multi_services
=== RUN   TestValidateFile/deploy_custom_agent
=== RUN   TestValidateFile/deploy_custom_agent_invalid_property
    --- PASS: TestValidateFile/deploy_custom_agent_multi_services (0.03s)
    --- PASS: TestValidateFile/deploy_custom_agent (0.03s)
    --- PASS: TestValidateFile/deploy_custom_agent_invalid_property (0.03s)
```

Reply:
Thanks!